Fails to build on 32 bit x86

There seems to be some invalid assumptions about the `long int *` size corresponding `int64`, which is not the case on 32 bit. gcc 14 catches this.

I'm not sure how to properly define `gint64`, `guint64_t` and `gsize` for cython?

```
Compiling gbinder.pyx because it changed.
[1/1] Cythonizing gbinder.pyx
gbinder.c: In function '__pyx_pf_7gbinder_11RemoteReply_10read_int64':
gbinder.c:8784:75: error: passing argument 2 of 'gbinder_remote_reply_read_int64' from incompatible pointer type [-Wincompatible-pointer-types]
 8784 |   __pyx_v_status = gbinder_remote_reply_read_int64(__pyx_v_self->_reply, (&__pyx_v_value));
      |                                                                          ~^~~~~~~~~~~~~~~
      |                                                                           |
      |                                                                           long int *
In file included from gbinder.c:1254:
/usr/include/gbinder/gbinder_remote_reply.h:73:13: note: expected 'gint64 *' {aka 'long long int *'} but argument is of type 'long int *'
   73 |     gint64* value);
      |     ~~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_11RemoteReply_12read_uint64':
gbinder.c:8931:76: error: passing argument 2 of 'gbinder_remote_reply_read_uint64' from incompatible pointer type [-Wincompatible-pointer-types]
 8931 |   __pyx_v_status = gbinder_remote_reply_read_uint64(__pyx_v_self->_reply, (&__pyx_v_value));
      |                                                                           ~^~~~~~~~~~~~~~~
      |                                                                            |
      |                                                                            long unsigned int *
/usr/include/gbinder/gbinder_remote_reply.h:78:14: note: expected 'guint64 *' {aka 'long long unsigned int *'} but argument is of type 'long unsigned int *'
   78 |     guint64* value);
      |     ~~~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_13RemoteRequest_20read_int64':
gbinder.c:19145:75: error: passing argument 2 of 'gbinder_remote_request_read_int64' from incompatible pointer type [-Wincompatible-pointer-types]
19145 |   __pyx_v_status = gbinder_remote_request_read_int64(__pyx_v_self->_req, (&__pyx_v_value));
      |                                                                          ~^~~~~~~~~~~~~~~
      |                                                                           |
      |                                                                           long int *
In file included from gbinder.c:1255:
/usr/include/gbinder/gbinder_remote_request.h:95:13: note: expected 'gint64 *' {aka 'long long int *'} but argument is of type 'long int *'
   95 |     gint64* value);
      |     ~~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_13RemoteRequest_22read_uint64':
gbinder.c:19292:76: error: passing argument 2 of 'gbinder_remote_request_read_uint64' from incompatible pointer type [-Wincompatible-pointer-types]
19292 |   __pyx_v_status = gbinder_remote_request_read_uint64(__pyx_v_self->_req, (&__pyx_v_value));
      |                                                                           ~^~~~~~~~~~~~~~~
      |                                                                            |
      |                                                                            long unsigned int *
/usr/include/gbinder/gbinder_remote_request.h:100:14: note: expected 'guint64 *' {aka 'long long unsigned int *'} but argument is of type 'long unsigned int *'
  100 |     guint64* value);
      |     ~~~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_6Reader_10read_int64':
gbinder.c:32252:73: error: passing argument 2 of 'gbinder_reader_read_int64' from incompatible pointer type [-Wincompatible-pointer-types]
32252 |   __pyx_v_status = gbinder_reader_read_int64((&__pyx_v_self->_reader), (&__pyx_v_value));
      |                                                                        ~^~~~~~~~~~~~~~~
      |                                                                         |
      |                                                                         long int *
In file included from gbinder.c:1252:
/usr/include/gbinder/gbinder_reader.h:101:13: note: expected 'gint64 *' {aka 'long long int *'} but argument is of type 'long int *'
  101 |     gint64* value);
      |     ~~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_6Reader_12read_uint64':
gbinder.c:32368:74: error: passing argument 2 of 'gbinder_reader_read_uint64' from incompatible pointer type [-Wincompatible-pointer-types]
32368 |   __pyx_v_status = gbinder_reader_read_uint64((&__pyx_v_self->_reader), (&__pyx_v_value));
      |                                                                         ~^~~~~~~~~~~~~~~
      |                                                                          |
      |                                                                          long unsigned int *
/usr/include/gbinder/gbinder_reader.h:106:14: note: expected 'guint64 *' {aka 'long long unsigned int *'} but argument is of type 'long unsigned int *'
  106 |     guint64* value);
      |     ~~~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_6Reader_30read_hidl_vec':
gbinder.c:33430:75: error: passing argument 2 of 'gbinder_reader_read_hidl_vec' from incompatible pointer type [-Wincompatible-pointer-types]
33430 |   __pyx_v_value = gbinder_reader_read_hidl_vec((&__pyx_v_self->_reader), (&__pyx_v_count), (&__pyx_v_elemsize));
      |                                                                          ~^~~~~~~~~~~~~~~
      |                                                                           |
      |                                                                           long unsigned int *
/usr/include/gbinder/gbinder_reader.h:161:12: note: expected 'gsize *' {aka 'unsigned int *'} but argument is of type 'long unsigned int *'
  161 |     gsize* count,
      |     ~~~~~~~^~~~~
gbinder.c:33430:93: error: passing argument 3 of 'gbinder_reader_read_hidl_vec' from incompatible pointer type [-Wincompatible-pointer-types]
33430 |   __pyx_v_value = gbinder_reader_read_hidl_vec((&__pyx_v_self->_reader), (&__pyx_v_count), (&__pyx_v_elemsize));
      |                                                                                            ~^~~~~~~~~~~~~~~~~~
      |                                                                                             |
      |                                                                                             long unsigned int *
/usr/include/gbinder/gbinder_reader.h:162:12: note: expected 'gsize *' {aka 'unsigned int *'} but argument is of type 'long unsigned int *'
  162 |     gsize* elemsize);
      |     ~~~~~~~^~~~~~~~
gbinder.c: In function '__pyx_pf_7gbinder_6Reader_32read_hidl_vec1':
gbinder.c:33603:76: error: passing argument 2 of 'gbinder_reader_read_hidl_vec1' from incompatible pointer type [-Wincompatible-pointer-types]
33603 |   __pyx_v_value = gbinder_reader_read_hidl_vec1((&__pyx_v_self->_reader), (&__pyx_v_count), __pyx_v_expected_elemsize);
      |                                                                           ~^~~~~~~~~~~~~~~
      |                                                                            |
      |                                                                            long unsigned int *
/usr/include/gbinder/gbinder_reader.h:167:12: note: expected 'gsize *' {aka 'unsigned int *'} but argument is of type 'long unsigned int *'
  167 |     gsize* count,
      |     ~~~~~~~^~~~~
gbinder.c: In function '__pyx_pf_7gbinder_6Reader_48read_byte_array':
gbinder.c:34472:77: error: passing argument 2 of 'gbinder_reader_read_byte_array' from incompatible pointer type [-Wincompatible-pointer-types]
34472 |   __pyx_v_value = gbinder_reader_read_byte_array((&__pyx_v_self->_reader), (&__pyx_v_len));
      |                                                                            ~^~~~~~~~~~~~~
      |                                                                             |
      |                                                                             long unsigned int *
/usr/include/gbinder/gbinder_reader.h:227:12: note: expected 'gsize *' {aka 'unsigned int *'} but argument is of type 'long unsigned int *'
  227 |     gsize* len); /* Since 1.0.12 */
      |     ~~~~~~~^~~
error: command '/usr/lib/ccache/bin/gcc' failed with exit code 1
```

--- a/cgbinder.pxd
+++ b/cgbinder.pxd
@@ -40,6 +40,8 @@ cdef extern from "gbinder/gbinder_types.h":
 
     ctypedef GBinderLocalReply* (*GBinderLocalTransactFunc)(GBinderLocalObject* obj, GBinderRemoteRequest* req, unsigned int code, unsigned int flags, int* status, void* user_data)
 
+from libc.stdint cimport int64_t, uint64_t
+
 cdef extern from "gbinder/gbinder_servicemanager.h":
     ctypedef void (*GBinderServiceManagerFunc)(GBinderServiceManager* sm, void* user_data)
     ctypedef bint (*GBinderServiceManagerListFunc)(GBinderServiceManager* sm, char** services, void* user_data)
@@ -162,8 +164,8 @@ cdef extern from "gbinder/gbinder_reader.h":
     bint gbinder_reader_read_bool(GBinderReader* reader, bint* value)
     bint gbinder_reader_read_int32(GBinderReader* reader, signed int* value)
     bint gbinder_reader_read_uint32(GBinderReader* reader, unsigned int* value)
-    bint gbinder_reader_read_int64(GBinderReader* reader, signed long* value)
-    bint gbinder_reader_read_uint64(GBinderReader* reader, unsigned long* value)
+    bint gbinder_reader_read_int64(GBinderReader* reader, int64_t* value)
+    bint gbinder_reader_read_uint64(GBinderReader* reader, uint64_t* value)
     bint gbinder_reader_read_float(GBinderReader* reader, float* value)
     bint gbinder_reader_read_double(GBinderReader* reader, double* value)
     int gbinder_reader_read_fd(GBinderReader* reader)
@@ -174,8 +176,8 @@ cdef extern from "gbinder/gbinder_reader.h":
     GBinderBuffer* gbinder_reader_read_buffer(GBinderReader* reader)
 
     const void* gbinder_reader_read_hidl_struct1(GBinderReader* reader, unsigned long size)
-    const void* gbinder_reader_read_hidl_vec(GBinderReader* reader, unsigned long* count, unsigned long* elemsize)
-    const void* gbinder_reader_read_hidl_vec1(GBinderReader* reader, unsigned long* count, unsigned int expected_elemsize)
+    const void* gbinder_reader_read_hidl_vec(GBinderReader* reader, size_t* count, size_t* elemsize)
+    const void* gbinder_reader_read_hidl_vec1(GBinderReader* reader, size_t* count, unsigned int expected_elemsize)
     char* gbinder_reader_read_hidl_string(GBinderReader* reader)
     const char* gbinder_reader_read_hidl_string_c(GBinderReader* reader)
     char** gbinder_reader_read_hidl_string_vec(GBinderReader* reader)
@@ -188,7 +190,7 @@ cdef extern from "gbinder/gbinder_reader.h":
     #bint gbinder_reader_read_nullable_string16_utf16(GBinderReader* reader, const unsigned short** out, unsigned long* len)
     #const unsigned short* gbinder_reader_read_string16_utf16(GBinderReader* reader,unsigned long* len)
     bint gbinder_reader_skip_string16(GBinderReader* reader)
-    const void* gbinder_reader_read_byte_array(GBinderReader* reader, unsigned long* len)
+    const void* gbinder_reader_read_byte_array(GBinderReader* reader, size_t* len)
     unsigned long gbinder_reader_bytes_read(const GBinderReader* reader)
     unsigned long gbinder_reader_bytes_remaining(const GBinderReader* reader)
     #void gbinder_reader_copy(GBinderReader* dest, const GBinderReader* src)
@@ -212,8 +214,8 @@ cdef extern from "gbinder/gbinder_remote_reply.h":
 
     bint gbinder_remote_reply_read_int32(GBinderRemoteReply* reply, signed int* value)
     bint gbinder_remote_reply_read_uint32(GBinderRemoteReply* reply, unsigned int* value)
-    bint gbinder_remote_reply_read_int64(GBinderRemoteReply* reply, signed long* value)
-    bint gbinder_remote_reply_read_uint64(GBinderRemoteReply* reply, unsigned long* value)
+    bint gbinder_remote_reply_read_int64(GBinderRemoteReply* reply, int64_t* value)
+    bint gbinder_remote_reply_read_uint64(GBinderRemoteReply* reply, uint64_t* value)
     const char* gbinder_remote_reply_read_string8(GBinderRemoteReply* reply)
     char* gbinder_remote_reply_read_string16(GBinderRemoteReply* reply)
     GBinderRemoteObject* gbinder_remote_reply_read_object(GBinderRemoteReply* reply)
@@ -232,8 +234,8 @@ cdef extern from "gbinder/gbinder_remote_request.h":
 
     bint gbinder_remote_request_read_int32(GBinderRemoteRequest* req, signed int* value)
     bint gbinder_remote_request_read_uint32(GBinderRemoteRequest* req, unsigned int* value)
-    bint gbinder_remote_request_read_int64(GBinderRemoteRequest* req, signed long* value)
-    bint gbinder_remote_request_read_uint64(GBinderRemoteRequest* req, unsigned long* value)
+    bint gbinder_remote_request_read_int64(GBinderRemoteRequest* req, int64_t* value)
+    bint gbinder_remote_request_read_uint64(GBinderRemoteRequest* req, uint64_t* value)
     const char* gbinder_remote_request_read_string8(GBinderRemoteRequest* req)
     char* gbinder_remote_request_read_string16(GBinderRemoteRequest* req)
     GBinderRemoteObject* gbinder_remote_request_read_object(GBinderRemoteRequest* self)

--- a/gbinder.pyx
+++ b/gbinder.pyx
@@ -68,6 +68,8 @@ cdef class RemoteObject:
 cdef void remote_object_local_notify_func(cgbinder.GBinderRemoteObject* obj, void* user_data) noexcept with gil:
     (<object>user_data).notify_func_callback()
 
+from libc.stdint cimport int64_t, uint64_t
+
 cdef class RemoteReply:
     cdef cgbinder.GBinderRemoteReply* _reply
 
@@ -114,7 +116,7 @@ cdef class RemoteReply:
     def read_int64(self):
         if self._reply is NULL:
             return None
-        cdef signed long value
+        cdef int64_t value
         cdef bint status
         status = cgbinder.gbinder_remote_reply_read_int64(self._reply, &value)
         return status, value
@@ -122,7 +124,7 @@ cdef class RemoteReply:
     def read_uint64(self):
         if self._reply is NULL:
             return None
-        cdef unsigned long value
+        cdef uint64_t value
         cdef bint status
         status = cgbinder.gbinder_remote_reply_read_uint64(self._reply, &value)
         return status, value
@@ -441,7 +443,7 @@ cdef class RemoteRequest:
     def read_int64(self):
         if self._req is NULL:
             return None
-        cdef signed long value
+        cdef int64_t value
         cdef bint status
         status = cgbinder.gbinder_remote_request_read_int64(self._req, &value)
         return status, value
@@ -449,7 +451,7 @@ cdef class RemoteRequest:
     def read_uint64(self):
         if self._req is NULL:
             return None
-        cdef unsigned long value
+        cdef uint64_t value
         cdef bint status
         status = cgbinder.gbinder_remote_request_read_uint64(self._req, &value)
         return status, value
@@ -836,13 +838,13 @@ cdef class Reader:
         return status, value
 
     def read_int64(self):
-        cdef signed long value
+        cdef int64_t value
         cdef bint status
         status = cgbinder.gbinder_reader_read_int64(&self._reader, &value)
         return status, value
 
     def read_uint64(self):
-        cdef unsigned long value
+        cdef uint64_t value
         cdef bint status
         status = cgbinder.gbinder_reader_read_uint64(&self._reader, &value)
         return status, value
@@ -890,12 +892,12 @@ cdef class Reader:
         return <object>value
 
     def read_hidl_vec(self):
-        cdef unsigned long count, elemsize
+        cdef size_t count, elemsize
         cdef const void* value = cgbinder.gbinder_reader_read_hidl_vec(&self._reader, &count, &elemsize)
         return <object>value, count, elemsize
 
     def read_hidl_vec1(self, unsigned int expected_elemsize):
-        cdef unsigned long count
+        cdef size_t count
         cdef const void* value = cgbinder.gbinder_reader_read_hidl_vec1(&self._reader, &count, expected_elemsize)
         return <object>value, count
 
@@ -931,7 +933,7 @@ cdef class Reader:
         return cgbinder.gbinder_reader_skip_string16(&self._reader)
 
     def read_byte_array(self):
-        cdef unsigned long len
+        cdef size_t len
         cdef const void* value = cgbinder.gbinder_reader_read_byte_array(&self._reader, &len)
         return <object>value, len
 

